app/knights/knight.py

```diff
@@ -0,0 +1,25 @@
+class Knight:
+
+    def __init__(self, config: dict) -> None:
+
+        self.name = config.get("name")
+        self.power = config.get("power")
+        self.hp = config.get("hp")
+        self.protection = 0
+
+        if config["armour"] and len(config["armour"]) > 0:

```

Comment:
The code assumes that 'armour' is always present in the config dictionary. Consider using `config.get('armour', [])` to avoid a `KeyError` if 'armour' is missing.